src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItem.scala

```diff
@@ -69,69 +70,101 @@ object EasyStageFileItem {
-      _                <- items.last match {case (sdo, _, parentRelation) => createFileSdo(sdo, parentRelation) }
+      existingAncestor <- s.easyFilesAndFolders.getExistingAncestor(s.pathInDataset.get, datasetId)
+      _                = createFolderSdos(existingAncestor, s.pathInDataset.get, datasetSdoSetDir)
+      _                <- createFileSdoForExistingDataset(datasetSdoSetDir, existingAncestor)
     } yield ()
   }
 
-  def relPath(parentPath: String, path: String): String =
-    if (parentPath.isEmpty) new File(path).toString // prevent a leading slash
-    else new File(parentPath, path).toString
-
-  def getPathElements()(implicit s: FileItemSettings): Try[(String, String, Seq[String])] = {
-    val file = s.pathInDataset.get
-    s.easyFilesAndFolders.getExistingAncestor(file, s.datasetId.get)
-      .map { case (parentPath, parentId) =>
-        log.debug(s"Parent in repository: $parentId $parentPath")
-        val newItems = file.toString.replaceFirst(s"^$parentPath/", "").split("/")
-        (parentId, parentPath, newItems.toSeq)
-      }
-  }
-
-  def getItemsToStage(pathElements: Seq[String], datasetSdoSet: File, existingFolderId: String): Seq[(File, String, (String, String))] = {
+  def getItemsToStage(pathElements: Seq[String],
+                      datasetSdoSet: File,
+                      existingFolderId: String
+                     ): Seq[(File, String, RelationObject)] = {
+    trace(pathElements, datasetSdoSet, existingFolderId)
     getPaths(pathElements)
-    .foldLeft(Seq[(File, String, (String, String))]())((items, path) => {
+    .foldLeft(Seq[(File, String, RelationObject)]())((items, path) => {
       items match {
-        case s@Seq() => s :+ (new File(datasetSdoSet, toSdoName(path)), path, "object" -> s"info:fedora/$existingFolderId")
+        case s@Seq() =>
+          val sdoDir = new File(datasetSdoSet, toSdoName(path))
+          log.debug(s"$sdoDir $path")
+          s :+ (sdoDir, path, "object" -> s"info:fedora/$existingFolderId")
         case seq =>
           val parentFolderSdoName = seq.last match { case (sdo, _,  _) => sdo.getName}
-          seq :+ (new File(datasetSdoSet, toSdoName(path)), path, "objectSDO" -> parentFolderSdoName)
+          val sdoDir = new File(datasetSdoSet, toSdoName(path))
+          log.debug(s"$sdoDir $path")
+          seq :+ (sdoDir, path, "objectSDO" -> parentFolderSdoName)
       }
     })
   }
 
-  def getPaths(path: Seq[String]): Seq[String] =
+  private def getPaths(path: Seq[String]): Seq[String] = {
+    trace(path)
     if(path.isEmpty) Seq()
     else path.tail.scanLeft(path.head)((acc, next) => s"$acc/$next")
+  }
 
+  private def createFolderSdos(existingAncestor: ExistingAncestor,
+                               file: File,
+                               datasetSdoSetDir: File
+                              )(implicit s: FileItemSettings) = {
+    val (existingPath, _) = existingAncestor
+
+    @tailrec
+    def createParent(child: File): Unit = {
```

Comment:
I don't like this method implementation with the manual null checks, but I can't yet think of a better way, since you want to keep the tail recursiveness... Maybe some kind of fold?